rules-java/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/GatherIgnoredFileNamesRuleProvider.java

```diff
@@ -55,72 +56,70 @@ public RulePhase getPhase()
-        for (FileModel ignoredRegexesFileModel : payload.getUserIgnorePaths())
+        for (FileModel ignoredRegexesFileModel : windupConfigM.getUserIgnorePaths())
         {
 
-            if (ignoredRegexesFileModel.isDirectory())
+            // Files pointed to by user are read regardless its name.
+            if (!ignoredRegexesFileModel.isDirectory()){
+                filesUrl.add(Paths.get(ignoredRegexesFileModel.getFilePath()));
+                continue;
+            }
+
+            // Directories are searched for files ending with IGNORE_FILE_EXTENSION ("windup-ignore.txt").
+            try
             {
-                try
+                Files.walkFileTree(Paths.get(ignoredRegexesFileModel.getFilePath()), new SimpleFileVisitor<Path>()
                 {
-                    Files.walkFileTree(Paths.get(ignoredRegexesFileModel.getFilePath()), new SimpleFileVisitor<Path>()
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException
                     {
-                        @Override
-                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException
-                        {
-                            if (file.getFileName().toString().toLowerCase().endsWith(IGNORE_FILE_EXTENSION))
-                            {
-                                filesUrl.add(file);
-                            }
-                            return FileVisitResult.CONTINUE;
-                        }
-                    });
-
-                }
-                catch (IOException e1)
-                {
-                    log.warning("IOException thrown when trying to access the ignored file regexes in " + ignoredRegexesFileModel.getFilePath());
-                }
+                        if (file.getFileName().toString().toLowerCase().endsWith(IGNORE_FILE_EXTENSION))
+                            filesUrl.add(file);
+
+                        return FileVisitResult.CONTINUE;
+                    }
+                });
+
             }
-            else
+            catch (IOException ex)
             {
-                filesUrl.add(Paths.get(ignoredRegexesFileModel.getFilePath()));
+                log.log(Level.WARNING, "Failed reading the ignored file regexes in "
+                        + ignoredRegexesFileModel.getFilePath() + ex.getMessage(), ex);
             }
         }
+
+        // Load regexes from the files found.
         for (Path filePath : filesUrl)
         {
             readAndAddFileRegexes(filePath, javaCfg, event.getGraphContext());
         }
     }
 
-    private void readAndAddFileRegexes(Path filePath, WindupJavaConfigurationModel javaCfg, GraphContext context)
+
+    private void readAndAddFileRegexes(Path filePath, WindupJavaConfigurationModel javaCfg, GraphContext gCtx)
     {
         File file = filePath.toFile();
-        if (file.exists())
+        if (!file.exists())
```

Comment:
I don't agree with Martin Fowler's "guard clauses" approach in simple cases where one or few checks are being made. I find that it makes the code "less" clear, so let's try to avoid refactoring to this pattern too frequently. If there is a portion of code that you feel could benefit from this refactoring, let's discuss it first.